server/datastore/mysql/migrations/tables/20171116163618_CreateTableOsqueryOptions.go

```diff
@@ -15,7 +25,137 @@ func Up_20171116163618(tx *sql.Tx) error {
 		"PRIMARY KEY (`id`)" +
 		") ENGINE=InnoDB DEFAULT CHARSET=utf8;"
 	_, err := tx.Exec(sqlStatement)
-	return err
+	if err != nil {
+		return errors.Wrap(err, "create table osquery_options")
+	}
+
+	// Check whether there are existing options to migrate, or we should insert
+	// a new default set of options
+	var count int
+	err = tx.QueryRow("SELECT count(1) FROM options").Scan(&count)
+	if err != nil {
+		return errors.Wrap(err, "get options count")
+	}
+
+	if count > 0 {
+		// Migrate existing options
+		err = migrateOptions(tx)
+		if err != nil {
+			return errors.Wrap(err, "migrate options")
+		}
+	} else {
+		// Insert default options
+		_, err = tx.Exec("INSERT INTO `osquery_options`" +
+			"(override_type, override_identifier, options)" +
+			`VALUES (0, "", '{"options": {"logger_plugin": "tls", "pack_delimiter": "/", "logger_tls_period": 10, "distributed_plugin": "tls", "disable_distributed": false, "logger_tls_endpoint": "/api/v1/osquery/log", "distributed_interval": 10, "distributed_tls_max_attempts": 3}, "decorators": {"load": ["SELECT uuid AS host_uuid FROM system_info;", "SELECT hostname AS hostname FROM system_info;"]}}')`,
+		)
+		if err != nil {
+			return errors.Wrap(err, "insert options")
+		}
+	}
+
+	return nil
+}
+
+func migrateOptions(tx *sql.Tx) error {
+	type configForExport struct {
```

Comment:
This is just moved from https://github.com/kolide/fleet/pull/2258/files#diff-b49dfacbb915d42e49cab4e6b35cad44L26